### PR TITLE
Added Python 3.12

### PR DESCRIPTION
**Pull request by radarhere**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -25,6 +25,7 @@ jobs:
       - name: Install dependencies
         run: |
           sudo apt-get install libsane-dev
+          python3 -m pip install setuptools
```
Comment (hugovk) [inline .github/workflows/test.yml:28]: We shouldn't assume 3.12 users have setuptools installed, and should specify the build backend using pyproject.toml 

https://packaging.python.org/en/latest/tutorials/packaging-projects/

Comment (radarhere) [reply .github/workflows/test.yml]: Ok, I've replaced this with pyproject.toml

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+[build-system]
+requires = [
+  "setuptools>=67.8",
+]
```
Comment (hugovk) [inline pyproject.toml:4]: Let's be explicit about the backend:

* https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html
* https://packaging.python.org/en/latest/tutorials/packaging-projects/

```suggestion
[build-system]
requires = [
  "setuptools>=67.8",
]
build-backend = "setuptools.build_meta"
```

**Review by hugovk — APPROVED** [review]
🚀

**Comment by manisandro** [comment]
Thanks!

**Pull request merged**